pkg/cmd/get.go

```diff
@@ -0,0 +1,182 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+)
+
+type GetOptions struct {
+	AllNamespaces  bool
+	IsCoreResource bool
+	APIPath        string
+	Resource       string
+	Token          string
+
+	KubeArchiveHost string
+
+	RESTConfig *rest.Config
+	kubeFlags  *genericclioptions.ConfigFlags
+}
+
+func NewGetOptions() *GetOptions {
+	return &GetOptions{
+		kubeFlags:       genericclioptions.NewConfigFlags(true),
+		KubeArchiveHost: "https://localhost:8081",
+	}
+}
+
+func NewGetCmd() *cobra.Command {
+	o := NewGetOptions()
+
+	cmd := &cobra.Command{
+		Use:   "get [RESOURCES]",
+		Short: "Command to get resources from KubeArchive",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			err = o.Validate(args)
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete(args)
```

Comment:
It might be worth getting o.Complete to work as a [`cobra.Command.ValidArgsFunction`](https://pkg.go.dev/github.com/spf13/cobra#Command.ValidArgsFunction) ([see example](https://github.com/spf13/cobra/blob/main/site/content/completions/_index.md#dynamic-completion-of-nouns))

Reply:
`ValidArgsFunction` is supposed to return values for shell completion, while `Complete` fills out fields in options for later processing. I don't think its worth investing time here.